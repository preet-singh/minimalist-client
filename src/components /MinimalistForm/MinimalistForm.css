.Minimalist-form {
    background-color: #1d3656;
    color: #dedede;
    padding: 16px;
    max-width: 332px;
  }
  
  .Minimalist-form label {
    display: block;
    font-size: 16px;
    padding-bottom: 8px;
  }
  
  .Minimalist-form input,
  .Minimalist-form textarea,
  .Minimalist-form select {
    font-size: 14px;
    padding: 8px;
    width: 300px;
  }
  
  .Minimalist-form select {
    height: 33px;
  }
  
  .Minimalist-form > .field {
    margin-bottom: 16px;
  }
  .Minimalist-form > .buttons {
    margin-top: 26px;
  }
  
  .Minimalist-form button {
    background-color: #010d1e;
    border-radius: 4px;
    border: 0;
    color: #cacaca;
    display: block;
    font-size: 18px;
    font-weight: bold;
    padding: 8px;
    text-align: center;
    text-decoration: none;
  }
  .Minimalist-form button[type="submit"]:hover {
    background-color: #29862e;
    color: white;
  }
  