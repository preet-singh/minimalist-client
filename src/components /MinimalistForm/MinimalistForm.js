import React from 'react'
import './MinimalistForm.css'
import PropTypes from 'prop-types';

export default function MinimalistForm(props) {
  const { className, ...otherProps } = props
  return (
    <form
      className={['Minimalist-form', className].join(' ')}
      action='#'
      {...otherProps}
    />
  )
}

MinimalistForm.propTypes = {
  classname: PropTypes.string.isRequired,
  otherProps: PropTypes.object.isRequired
}
