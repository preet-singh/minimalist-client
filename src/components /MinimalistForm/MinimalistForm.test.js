import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import MinimalistForm from './MinimalistForm'

describe(`MinimalistForm component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  }

  it('renders a form.MinimalistForm by default', () => {
    const wrapper = shallow(<MinimalistForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the MinimalistForm given props', () => {
    const wrapper = shallow(<MinimalistForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
