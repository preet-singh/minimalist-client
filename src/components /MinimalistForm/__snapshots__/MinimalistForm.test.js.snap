// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinimalistForm component renders a form.MinimalistForm by default 1`] = `
<form
  action="#"
  className="Minimalist-form "
/>
`;

exports[`MinimalistForm component renders the MinimalistForm given props 1`] = `
<form
  action="#"
  className="Minimalist-form test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</form>
`;
